src/torrent.cpp

```diff
@@ -1855,30 +1855,25 @@ namespace libtorrent
 			{
 				std::vector<std::shared_ptr<torrent>> ts = m_ses.find_collection(c);
 
-				for (std::vector<std::shared_ptr<torrent>>::iterator k = ts.begin()
-					, end2(ts.end()); k != end2; ++k)
+				for (auto const& t : ts)
```

Comment:
sometimes I wonder how exactly lists of `std::shared_ptr` should be iterated, by ref, or by copy (just a comment)